### ISSUE TITLE
trailing_stop_positive doen't work as expected

### ISSUE THREAD

**Issue by farmage** [freqtrade/freqtrade#7655]
## Describe your environment

  * Operating system: Ubuntu 20.
  * Python Version: 3.8 (`python -V`)
  * CCXT version: 2.05 (`pip freeze | grep ccxt`)
  * Freqtrade Version: lastest dev (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## Your question

I'm observing unexpected behavior of trailing stop
my configuration is
leverage value is 20
"stoploss_on_exchange": true,
    "stoploss": "market",

```python
    stoploss = -0.3
 
    trailing_stop = True
    trailing_stop_positive = 0.1
    trailing_stop_positive_offset = 0.11
    trailing_only_offset_is_reached = True

    position_adjustment_enable = True
    max_entry_position_adjustment = 2
```
Adjustment code is
```python
        if current_profit >= 0.11 and trade.nr_of_successful_exits == 0:
            # print(f"{trade.pair}: Take partial profit 1st {trade.stake_amount / 2}")
            return -(trade.stake_amount*2 / 3)
```

So, the flow:
1. Price reached a profit of 11%. 
2. The adjustment code closes partial profit
3. After closing part of the position, I expect to shift the stop loss to break even price, as specified in the above config, but the stop loss remains on the same level, at -30% 
I've noticed the same behavior in live and dry run modes.

Is It a bug or do I not understand trailing stop calculation in freqtrade?
UPD:
This only happens if the price quickly goes in the opposite direction. For example:
1. The price reached 11% level of profit. 
2. FQ sends "market" exit order for part of position
3. At the time of exchange closes part of position the profit is 9%

Maybe this is the reason the stop loss didn't move

**Comment by xmatthias** [comment]
you're taking part of the profit - while that order is open, stoploss won't readjust.

once that order closes, stoploss will readjust using your settings - but as you correctly point out - profit is now < 11% - so it's no longer a valid scenario for adjustment.

you can probably get around this by using a custom stoploss (instead of trailing_stop) - and checking for `trade.nr_of_successful_exits > 0` ... or something similar.

**Comment by farmage** [comment]
Thanks! custom_stoploss seems to be a good solution. I've made changes to the code and started testing the bot.
